sprint_7-solution-in-file-manager

Enfurecid0 opened the pull request.

Enfurecid0 committed:
sprint_7-solution-in-file-manager

Enfurecid0 committed:
sprint_7-solution-in-file-manager

slavrikk commented on src/manager/InMemoryTaskManager.java:
```diff
@@ -11,12 +11,27 @@
-    private final Map<Integer, Task> tasks = new HashMap<>();
-    private final Map<Integer, Epic> epics = new HashMap<>();
-    private final Map<Integer, Subtask> subtasks = new HashMap<>();
+    protected static final Map<Integer, Task> tasks = new HashMap<>();
```
Лучше убрать `static` для всех полей коллекций, чтобы у каждого объекта класса менеджер были свои данные

slavrikk commented on src/manager/FileBackedTaskManager.java:
```diff
@@ -0,0 +1,167 @@
+        this.file = file;
+    }
+
+    public void save() {
```
Лучше private, так как это внутренняя логика класса

slavrikk commented on src/manager/InMemoryTaskManager.java:
```diff
@@ -11,12 +11,27 @@
+    protected int generatorId = 0;
+
+    @Override
+    public ArrayList<Task> getAllTasks() {
```
Лучше в качестве возвращающегося типа использовать интерфейсы:
`List<Task> getAllTasks()` , замечание для всех методов

slavrikk commented on src/manager/TaskManager.java:
```diff
@@ -4,10 +4,17 @@
 
 public interface TaskManager {
 
+    ArrayList<Task> getAllTasks();
```
Лучше `List<Task> getAllTasks();`

Enfurecid0 committed:
InMemoryTaskManager - убрать static.
FileBackedTaskManager - сделать методод save private.
TaskManager - использовать ссылку на интерфейс.
